yarn-project/sequencer-client/src/publisher/l1-publisher.ts

```diff
@@ -235,17 +236,19 @@ export class L1Publisher implements L2BlockReceiver {
     this.interrupted = false;
   }
 
-  // TODO: Check fee distributor has at least 0.5 ETH.
-  // Related to https://github.com/AztecProtocol/aztec-packages/issues/1588
-  // eslint-disable-next-line require-await
-  private async checkFeeDistributorBalance(): Promise<boolean> {
-    return true;
-  }
-
-  // TODO: Fail if blockchainStatus.nextBlockNum > thisBlockNum.
-  // Related to https://github.com/AztecProtocol/aztec-packages/issues/1588
-  private checkNextL2BlockNum(_thisBlockNum: number): Promise<boolean> {
-    return Promise.resolve(true);
+  /**
+   * Verifies that the given value of start state hash equals that on the rollup contract
+   * @param startStateHash The start state hash of the block we wish to publish.
+   * @returns Boolean indicating if the hashes are equal.
+   */
+  private async checkStartStateHash(startStateHash: Buffer): Promise<boolean> {
+    const fromChain = await this.txSender.getCurrentStateHash();
+    const areSame = startStateHash.equals(fromChain);
+    if (!areSame) {
+      this.log(`CONTRACT STATE HASH: ${fromChain.toString('hex')}`);
+      this.log(`NEW BLOCK STATE HASH: ${startStateHash.toString('hex')}`);
+    }
```

Comment:
Were these logs just for debugging during dev, or are they meant to be there?

Reply:
Initially for debugging. But if someone does ever get an invalid state hash when attempting to publish it would probably be useful to have the info so I left them here.